### ISSUE TITLE
Infinite scroll for Ant Table

### ISSUE THREAD

**Issue by StallionV** [ant-design/ant-design#22928]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
This has been reported before and Ant has shown 3rd party solutions 
e.g.  https://github.com/Leonard-Li777/antd-table-infinity, but they really do not work well out of the box, are jittery and not fully maintained. 

Given that for years people have been used to infinite scroll now consider mobile devices for example you just pull from top or push up from bottom to get more content. 
This way is much more intuitive. Most grid and table based libraries have also moved to infinite scroll.

It would be very much appreciated if Ant design works towards adding an in-built infinite scroll feature as a variation along with the pagination feature.

### What does the proposed API look like?
pagination: true || false || infinite

**Comment by RealDyllon** [comment]
See issue #11933, which refers to this PR: https://github.com/react-component/table/pull/231
The PR is still open :/

I found this fork, but I can't read the docs: https://github.com/Leonard-Li777/antd-table-infinity

**Comment by afc163** [comment]
#5904
#18062

**Comment by StallionV** [comment]
> See issue #11933, which refers to this PR: [react-component/table#231](https://github.com/react-component/table/pull/231)
> The PR is still open :/
> 
> I found this fork, but I can't read the docs: https://github.com/Leonard-Li777/antd-table-infinity

@afc163 can you help with merging this request or comment, looks like it has now been open for a year.

**Comment by StallionV** [comment]
> I found this fork, but I can't read the docs: https://github.com/Leonard-Li777/antd-table-infinity

The documentation is also in English but the library does not work well for all row heights.
The scroll has to be just based on detecting the scroll position since row heights can really vary per use case

**Comment by WangYang-Rex** [comment]
table的虚拟window 能否支持固定列以及rowSelection，官方给的例子只是实现了最最基础的table

**Comment by WangYang-Rex** [comment]
@afc163 上面的问题可以实现么，how?

**Comment by MaximusFT** [comment]
https://github.com/Leonard-Li777/antd-table-infinity
Unfortunately, this repo has many bugs.

But I adaptive this solution for an infinity scroll loader.
https://github.com/wubostc/virtualized-table-for-antd

Create component `SmartTable`, and use insted `import Table from 'antd/lib/table';`

```javascript
import React from 'react';
import PropTypes from 'prop-types';
import Table from 'antd/lib/table';
import { useVT } from 'virtualizedtableforantd4';

const scrollConfig = {
  y: 500,
  x: true,
};

const SmartTable = ({
  debug = false,
  infinity = false,
  dataSource = [],
  scroll = scrollConfig,
  onFetch = () => null,
  lastId,
  ...props
}) => {
  if (!infinity) {
    return <Table dataSource={dataSource} {...props} />;
  }

  const [vt] = useVT(
    () => ({
      onScroll: ({ isEnd }) => {
        if (isEnd) {
          console.log('loadDataByChunk');
          onFetch();
        }
      },
      scroll: { y: scrollConfig.y },
      debug,
    }),
    [lastId, dataSource]
  );

  return (
    <Table dataSource={dataSource} scroll={scroll} components={vt} pagination={false} {...props} />
  );
};

SmartTable.propTypes = {
  lastId: PropTypes.any,
  dataSource: PropTypes.any,
  debug: PropTypes.bool,
  infinity: PropTypes.bool,
  loadingIndicator: PropTypes.bool,
  onFetch: PropTypes.func,
  scroll: PropTypes.any,
};

export default SmartTable;
```

**Comment by douglasmamuty** [comment]
any update on this ?

**Comment by androidfanatic** [comment]
+1

**Comment by Alexander3** [comment]
+1

**Comment by iCodist** [comment]
+1

**Comment by Awesome94** [comment]
+1

**Comment by afc163** [comment]
Duplicate of https://github.com/ant-design/ant-design/issues/20339

**Comment by KajaPiechowka** [comment]
+1

**Comment by Alexander3** [comment]
@afc163  I don't understand connection between infinite scroll and feature you mentioned. You closed this one so will infinite scroll pagination style will be implemented as part of  [#20339](https://github.com/ant-design/ant-design/issues/20339) ?

**Comment by sujanpatro-surya** [comment]
@afc163 Any updates on whether infinite-scrolling will be supported or not? If not, can you please provide the best solution on how to achieve this?
https://github.com/ant-design/ant-design/issues/22928#issuecomment-1133059211 - didn't do any due diligence on this issue.
